src/core/components/hv-doc/index.tsx

```diff
@@ -39,12 +40,36 @@ export default class HvDocState extends PureComponent<
       styles: null,
       url: null,
     };
+    this.backBehaviorElementCache = [];
   }
 
   render(): React.ReactNode {
     return (
-      <Contexts.DocStateContext.Provider
+      <Contexts.DocContext.Provider
         value={{
+          backBehaviorElements: {
+            add: (elements: Element[]): void => {
+              if (elements?.length > 0) {
+                elements.forEach(e => {
+                  this.backBehaviorElementCache.push(e);
+                });
```

Comment:
I think the check for length > 0 is not needed, and the expression can be simplified to:
```suggestion
              this.backBehaviorElementCache.push(...elements);
```

Reply:
Thanks, I meant to go back and look for a cleaner way to implement this push